flocker/node/agents/blockdevice.py

```diff
@@ -1598,8 +1621,15 @@ def is_existing_block_device(dataset_id, path):
                     # in the filesystem yet.
                     pass
 
+        if ICloudAPI.providedBy(api):
+            live_instances = api.list_live_nodes()
+        else:
+            # Assume everything is alive:
+            live_instances = [volume.attached_to for volume in volumes
+                              if volume.attached_to is not None]
```

Comment:
This seems brittle to me.  If someone uses live_instances for anything other than checking volume.attached_to they might be in for a surprise.

In fact if this node doesn't have any volumes attached then `self.compute_instance_id() in live_instances` is False o.O

Perhaps instead have `live_instances == None` indicate that we do not know which instances are live, and set it to None here? Then add a method to RawState:

```
def is_instance_live(self, instance_id):
   if self.live_instances is None:
     return True
   return instance_id in self.live_instances
```
and use that below.